powerscale/provider/synciq_certificate_resource.go

```diff
@@ -0,0 +1,255 @@
+		Name:        types.StringValue(cert.Name),
+		Description: types.StringValue(cert.Description),
+		Path:        path,
+	}
+	return state, nil
+}
+
+// Read reads the resource state.
+func (r *SyncIQPeerCertificateResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var oldState models.SyncIQPeerCertificateResource
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &oldState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	state, err := r.get(ctx, oldState.ID.ValueString(), oldState.Path)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to read SyncIQ Peer Certificate", err.Error())
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+// Update updates the resource state.
+func (r *SyncIQPeerCertificateResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan models.SyncIQPeerCertificateResource
+	var oldState models.SyncIQPeerCertificateResource
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &oldState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// update the certificate
+	err := helper.UpdatePeerCert(ctx, r.client, oldState.ID.ValueString(), powerscale.V16CertificatesSyslogIdParams{
+		Name:        helper.GetKnownStringPointer(plan.Name),
+		Description: helper.GetKnownStringPointer(plan.Description),
+	})
+	if err != nil {
+		errStr := "Could not update syncIQ Peer Certificate with error: "
+		message := helper.GetErrorString(err, errStr)
+		resp.Diagnostics.AddError("Failed to update SyncIQ Peer Certificate", message)
+		return
+	}
+
+	// read the certificate
+	state, err := r.get(ctx, oldState.ID.ValueString(), oldState.Path)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get SyncIQ Peer Certificate after update", err.Error())
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+// Delete deletes the resource.
+func (r *SyncIQPeerCertificateResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state models.SyncIQPeerCertificateResource
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// delete cert
+	err := helper.DeletePeerCert(ctx, r.client, state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to delete SyncIQ Peer Certificate", err.Error())
+		return
+	}
+
+	resp.State.RemoveResource(ctx)
+}
+
+// ImportState imports the resource.
+func (r *SyncIQPeerCertificateResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	state := models.SyncIQPeerCertificateResource{
+		ID:          types.StringValue(req.ID),
+		Path:        "/dummy",
```

Comment:
Why these details are hardcoded?

Reply:
Path is only required during create. The array does not return it on subsequent reads. So writing dummy string to state.
As for name and description, they get updated when terraform refresh happens as part of post-import action.

Reply:
https://github.com/dell/terraform-provider-powerscale/pull/193/files#diff-b423b98401b6e6882a8db70fa093083eebcf80eafb50b68dd892c0f054af83b9R99-R112 tests that name and description gets updated